packages/compass-schema/src/stores/schema-analysis-reducer.ts

```diff
@@ -27,10 +25,15 @@ const DEFAULT_SAMPLE_SIZE = 1000;
 
 const ERROR_CODE_MAX_TIME_MS_EXPIRED = 50;
 
+export type SchemaAnalysisError = {
+  errorMessage: string;
+  errorType: 'TIMEOUT' | 'HIGH_COMPLEXITY' | 'GENERAL';
```

Comment:
nit, we usually use lower case for enum strings. Looks like they already were in upper case, mentioning it since we are making changes here anyway.